pkg/cli/clisqlclient/rows.go

```diff
@@ -78,18 +81,52 @@ func (r *sqlRows) Next(values []driver.Value) error {
 
 // NextResultSet prepares the next result set for reading.
 func (r *sqlRows) NextResultSet() (bool, error) {
-	if !r.rows.HasNextResultSet() {
-		return false, nil
-	}
-	return true, r.rows.NextResultSet()
+	return false, nil
 }
 
 func (r *sqlRows) ColumnTypeScanType(index int) reflect.Type {
-	return r.rows.ColumnTypeScanType(index)
+	o := r.rows.FieldDescriptions()[index].DataTypeOID
+	switch o {
+	case pgtype.Int8OID:
+		return reflect.TypeOf(int64(0))
+	case pgtype.Int4OID:
+		return reflect.TypeOf(int32(0))
+	case pgtype.Int2OID:
+		return reflect.TypeOf(int16(0))
+	case pgtype.VarcharOID, pgtype.TextOID:
+		return reflect.TypeOf("")
+	case pgtype.BoolOID:
+		return reflect.TypeOf(false)
+	case pgtype.DateOID, pgtype.TimeOID, 1266, pgtype.TimestampOID, pgtype.TimestamptzOID:
+		// 1266 is the OID for TimeTZ.
+		// TODO(rafi): Add TimetzOID to pgtype.
```

Comment:
i wonder whether time.Time is the appropriate thing to use for Time/TimeTZ. 24:00:00 time springs to mind as a point of confusion. that said, if we do that today then so be it.

can we use `pgtype.OID(oid.T_Timetz)` for now?